torch/csrc/utils/tensor_new.cpp

```diff
@@ -417,7 +417,7 @@ Tensor internal_new_from_data(
             " or an UntypedStorage, but got ",
             storage_scalar_type);
         tensor = at::empty(
-            sizes,
+            {0}, // sizes. Storage will be set later.
```

Comment:
are we relying on the `tensor.set_(...)` storage call to properly update the size metadata` on tensor instead? If so, can we add an assert that `tensor.sym_size() == sizes` after the set_() is done?